### ISSUE TITLE
Saved files disappear from the editor when using S3 storage

### ISSUE THREAD

**Issue by aku** [kestra-io/kestra#4077]
### Describe the issue

Steps to reproduce:
* Go to Editor tab in some flow
* Open files explorer panel
* Create and save or import some file
* Switch to another tab 
* Go back to editor
* File is gone

I've recorded a small video

![file_missing](https://github.com/kestra-io/kestra/assets/678596/c20a14a8-5a62-4e76-b56a-a6e9b4f0b711)

### Environment

- Kestra Version: 0.17.4
- Operating System (OS/Docker/Kubernetes): Ubuntu 12
- Java Version (if you don't run kestra in Docker):

**Comment by anna-geller** [comment]
more context https://kestra-io.slack.com/archives/C03FEC452NQ/p1718812771627879

**Comment by aku** [comment]
I've found the missing files in S3 storage:

```
2024-06-19 15:55            3  s3://kestra//test/_files/test.py
2024-06-19 18:26           11  s3://kestra//test/_files/test.txt
```

It looks like that the bug is in UI

**Comment by aku** [comment]
I've created a simple workflow that tries to read the file:

<img width="818" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/6636bd3f-5ac7-4969-ae8c-7a8299d6183c">

As you might notice the file is missing from UI, however the flow is able to access it

<img width="890" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/572f4fc8-d235-401c-91c8-150a2988aef8">

<img width="900" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/5aa060d2-d34e-4510-a308-278d99a36201">

If I try to create a file with the same name the system will overwrite the existing file (but the file is still not visible in UI)

**Comment by aku** [comment]
Another observation is that the files created via UI are missing from the exported files archive. It contains only yaml flows definitions.

**Comment by brian-mulier-p** [comment]
Hey ! I'll try to reproduce it later in the day. Can you try the same thing with a local storage (if not I'll reproduce it with S3 but it would help identifying whether it's a UI issue or backend one) ?

**Comment by aku** [comment]
@brian-mulier-p  I've tested this case with a dockerized kestra It does not reproduce with a local storage.

```bash
docker run --pull=always --rm -it -p 8080:8080 --user=root \
 -v /var/run/docker.sock:/var/run/docker.sock \
 -v /tmp:/tmp kestra/kestra:latest-full server local
```

<img width="329" alt="image" src="https://github.com/kestra-io/kestra/assets/678596/632d694c-2915-4c7c-be66-0d8da72b4d45">

**Comment by aku** [comment]
Also, I've checked export functionality and it was working correctly with a local dockerized kestra - test files were in the downloaded archive

**Comment by brian-mulier-p** [comment]
Thank's for providing effort with local storage, I'm setting up the reproducer :+1:

**Comment by aku** [comment]
> Thank's for providing effort with local storage, I'm setting up the reproducer 👍

To give more context:
I've started with Kestra 16.xx (i don't remember minor version but it was the last release before 17)
I've gradually upgraded my installation from 17.1 to 17.4 (updated binaries and re-installed plugins, I'm running Kestra in VMs)

I.e. if you will fail to reproduce it in 17.4 with a fresh install, it might be something related to migration from a previous versions. In this case I'm willing to do a local debugging/analysis if you provide me with ideas/instruction. Feel free to tag me in Slack

A file will be overwritten in S3 if I create a new one with the same name - it might be another/related bug.

**Comment by brian-mulier-p** [comment]
So yes I didn't manage to reproduce the issue. Dumb question but are you installing plugins on your own ? If so make sure your storage-s3 plugin is the 0.17.0 one.
Also can you show the listing content of your S3 bucket under /test/_files ? If it doesn't exist basically under /test

**Comment by brian-mulier-p** [comment]
Aside from that can you show your network tab (in your browser) when opening the editor ? Maybe export it in HAR format and provide it here. At least provide the /namespaces/test/files call output

**Comment by aku** [comment]
> So yes I didn't manage to reproduce the issue. Dumb question but are you installing plugins on your own ? If so make sure your storage-s3 plugin is the 0.17.0 one. Also can you show the listing content of your S3 bucket under /test/_files ? If it doesn't exist basically under /test

I have ansible playbook that installs plugins:

```yaml
    - name: Install kestra plugins # noqa command-instead-of-shell partial-become
      ansible.builtin.shell:
        cmd: '"{{ kestra_dir }}/kestra" plugins install --plugins="{{ kestra_dir }}/plugins"  "{{ item }}"'
      loop:
        - io.kestra.plugin:plugin-git:LATEST
        - io.kestra.plugin:plugin-fs:LATEST
        - io.kestra.plugin:plugin-elasticsearch:LATEST
        - io.kestra.plugin:plugin-kafka:LATEST
        - io.kestra.plugin:plugin-docker:LATEST
        - io.kestra.plugin:plugin-redis:LATEST
        - io.kestra.plugin:plugin-jdbc-postgres:LATEST
        - io.kestra.plugin:plugin-jdbc-clickhouse:LATEST
        - io.kestra.plugin:plugin-script-node:LATEST
        - io.kestra.plugin:plugin-script-python:LATEST
        - io.kestra.plugin:plugin-script-shell:LATEST
        - io.kestra.plugin:plugin-notifications:LATEST
        - io.kestra.plugin:plugin-serdes:LATEST
        - io.kestra.storage:storage-minio:LATEST
        - io.kestra.storage:storage-s3:LATEST
      become_user: "{{ kestra_sys_user }}"
      changed_when: false
      tags:
        - plugins
```

When a new version of Kestra is released I do the following:
* stop systemd services (I have multiple services for each Kestra component)
* download a fresh binary and update VMs
* run the code above to re-install plugins

```
❯ s3cmd  -c s3.cfg ls s3://kestra//test/_files/
                          DIR  s3://kestra//test/_files/test/
                          DIR  s3://kestra//test/_files/test2/
2024-06-19 22:39       DIROBJ  s3://kestra//test/_files/
2024-06-19 18:25        25141  s3://kestra//test/_files/WB.log
2024-06-19 22:39           20  s3://kestra//test/_files/test.py
2024-06-19 18:26           11  s3://kestra//test/_files/test.txt
```

All of these files/folders were created via UI but are missing in the editor

**Comment by aku** [comment]
> Aside from that can you show your network tab (in your browser) when opening the editor ? Maybe export it in HAR format and provide it here. At least provide the /namespaces/test/files call output

As I've mentioned in Slack there are no errors in dev console besides Chrome warnings on 3rd party cookies

I've recorded a video and a corresponding HAR file


https://github.com/kestra-io/kestra/assets/678596/d928dcd6-f726-4fa3-889c-799f41fbe1ae




[kestra.requests.har.zip](https://github.com/user-attachments/files/15917665/kestra.requests.har.zip)

**Comment by aku** [comment]
Since you mentioned plugins, I will try to manually delete and re-install s3 plugin. Maybe it will help

UPD:

I've logged into VMs, removed s3 plugins manually  (there were 16.x and 17.0 versions) and reinstalled the latest version. Then I've restarted kestra. Unfortunately, it did not help. Here is a list of plugins in the plugins directory:

```
plugin-docker-0.16.1.jar
plugin-docker-0.17.0.jar
plugin-elasticsearch-0.16.0.jar
plugin-elasticsearch-0.17.0.jar
plugin-fs-0.16.0.jar
plugin-fs-0.16.1.jar
plugin-fs-0.17.0.jar
plugin-git-0.16.1.jar
plugin-git-0.17.0.jar
plugin-jdbc-clickhouse-0.16.1.jar
plugin-jdbc-clickhouse-0.17.0.jar
plugin-jdbc-duckdb-0.16.1.jar
plugin-jdbc-duckdb-0.17.0.jar
plugin-jdbc-mysql-0.16.1.jar
plugin-jdbc-mysql-0.17.0.jar
plugin-jdbc-postgres-0.16.1.jar
plugin-jdbc-postgres-0.17.0.jar
plugin-kafka-0.16.0.jar
plugin-kafka-0.17.0.jar
plugin-malloy-0.16.1.jar
plugin-malloy-0.17.0.jar
plugin-notifications-0.16.0.jar
plugin-notifications-0.17.0.jar
plugin-redis-0.16.0.jar
plugin-redis-0.17.0.jar
plugin-script-groovy-0.16.1.jar
plugin-script-groovy-0.17.0.jar
plugin-script-julia-0.16.1.jar
plugin-script-julia-0.17.0.jar
plugin-script-jython-0.16.1.jar
plugin-script-jython-0.17.0.jar
plugin-script-nashorn-0.16.1.jar
plugin-script-nashorn-0.17.0.jar
plugin-script-node-0.16.1.jar
plugin-script-node-0.17.0.jar
plugin-script-python-0.16.1.jar
plugin-script-python-0.17.0.jar
plugin-script-r-0.16.1.jar
plugin-script-r-0.17.0.jar
plugin-script-shell-0.16.1.jar
plugin-script-shell-0.17.0.jar
plugin-serdes-0.16.0.jar
plugin-serdes-0.17.1.jar
plugins.txt
storage-minio-0.16.1.jar
storage-minio-0.17.0.jar
storage-s3-0.17.0.jar```

**Comment by brian-mulier-p** [comment]
For traceability:
the storage-minio was used directly against an S3 bucket (and not a minio one) which led to some issues among which the fact that in the storage-minio plugin we remove any leading slash when querying due to limitations from minio which in the end doesn't work properly with S3.

The right way is to use the storage-s3 plugin but in this case it was causing an issue as the DNS is not configured to reach [virtual-host-style S3 URL](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html). The behaviour of the S3 SDK is to detect the domain resolution styling by checking if the url contains any dot (mostly IP) prior to path. If it contains some it will default to path-styling else it goes to virtual-host-styling.

In the case above it's not working as we have a strong DNS limitation and the allowed styling is a fixed one that doesn't go with the above domain resolution rule.

To solve that I introduced a new [forcePathStyle configuration](https://github.com/kestra-io/storage-s3/commit/7366eb66970303989300e5a9889a4e0aef3a9bd0) for storage-s3 plugin. Putting it to true should solve the issue. Feel free to open again if it doesn't work.

Moreover I enhanced the [doc on storages](https://kestra.io/docs/configuration-guide/storage) to include [all properties currently available](https://github.com/kestra-io/docs/commit/3d1fe9a7ab7f9d990b16cd438259f8b5eb554abc)
